projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1600,7 +1616,10 @@ export class IgxTimePickerComponent implements
             displayVal = this._formatTime(this.value, this.format);
         }
 
-        this.displayValue = this.inputFormat.transform(displayVal);
+        // minor hack for preventing cursor jumping in IE
+        const normalizedValue = this.inputFormat.transform(displayVal);
+        this._renderer.setElementProperty(this.input.nativeElement, 'value', normalizedValue);
```

Comment:
Please update the displayValue private field here too. This will ensure that if someone is using the getter the value will be correct.